### PR TITLE
refactor(framework:skip) Merge `GrpcDriver` and `GrpcDriverStub`

### PR DESCRIPTION
**Pull request by panh99**

### CODE REVIEW

File: src/py/flwr/server/driver/grpc_driver.py [hunk 1]
```diff
@@ -53,153 +51,75 @@
 """
 
 
-class GrpcDriverStub:
-    """`GrpcDriverStub` provides access to the gRPC Driver API/service.
+class GrpcDriver(Driver):
+    """`Driver` class provides an interface to the Driver API.
```
Comment (danieljanes) [inline src/py/flwr/server/driver/grpc_driver.py:55]:
```suggestion
    """`GrpcDriver` provides an interface to the Driver API.
```

File: src/py/flwr/server/driver/grpc_driver.py [hunk 2]
```diff
@@ -53,153 +51,75 @@
...
 
     Parameters
     ----------
-    driver_service_address : Optional[str]
+    run_id : int
+        The identifier of the run.
+    driver_service_address : str (default: "[::]:9091")
         The IPv4 or IPv6 address of the Driver API server.
```
Comment (danieljanes) [inline src/py/flwr/server/driver/grpc_driver.py:62]:
```suggestion
        The address of the SuperLink Driver API service.
```

File: src/py/flwr/server/driver/grpc_driver.py [hunk 3]
```diff
@@ -53,153 +51,75 @@
...
-        self.root_certificates = root_certificates
-        self.channel: Optional[grpc.Channel] = None
-        self.stub: Optional[DriverStub] = None
+        self._run_id = run_id
+        self._run: Optional[Run] = None
+        self._addr = driver_service_address
+        self._cert = root_certificates
+        self._stub: Optional[DriverStub] = None
+        self._channel: Optional[grpc.Channel] = None
+        self.node = Node(node_id=0, anonymous=True)
```
Comment (danieljanes) [inline src/py/flwr/server/driver/grpc_driver.py:81]:
```suggestion
        self._run_id = run_id
        self._addr = driver_service_address
        self._cert = root_certificates
        self._run: Optional[Run] = None
        self._stub: Optional[DriverStub] = None
        self._channel: Optional[grpc.Channel] = None
        self.node = Node(node_id=0, anonymous=True)
```

File: src/py/flwr/server/driver/grpc_driver.py [hunk 4]
```diff
@@ -208,12 +128,24 @@ def _get_stub_and_run_id(self) -> Tuple[GrpcDriverStub, int]:
...
 
-        return self.stub, self._run.run_id
+    @property
+    def run(self) -> Run:
+        """Run information."""
+        self._init_run()
+        return Run(**vars(self._run))
+
+    @property
+    def stub(self) -> DriverStub:
```
Comment (danieljanes) [inline src/py/flwr/server/driver/grpc_driver.py:138]: This should be private (`_stub`). We do not want to expose `stub` as part of the public API of `GrpcDriver`, even though we use it in the `run_serverapp` module (for now, we will move `CreateRun` out of the Driver API soon).

Comment (panh99) [reply src/py/flwr/server/driver/grpc_driver.py]: That makes a lot of sense. Thanks for bringing it up. I've updated it.

File: src/py/flwr/server/driver/grpc_driver.py [hunk 5]
```diff
@@ -53,153 +51,75 @@
...
-    @property
-    def run(self) -> Run:
-        """Run information."""
-        self._get_stub_and_run_id()
-        return Run(**vars(cast(Run, self._run)))
-
-    def _get_stub_and_run_id(self) -> Tuple[GrpcDriverStub, int]:
+    def _init_run(self) -> None:
         # Check if is initialized
         if self._run is None:
```
[the remark this answers is not in the document]
Comment (panh99) [reply src/py/flwr/server/driver/grpc_driver.py]: Thanks! I've updated it.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**